### ISSUE TITLE
Using progress=True results in hung process after a previous run of sync process

### ISSUE THREAD

**Issue by tlvu** [bird-house/finch#121]
## Description

Running the following notebook, the async process never terminate
[finch_test-pavics.ipynb.txt](https://github.com/bird-house/finch/files/4451885/finch_test-pavics.ipynb.txt)

But restarting the Finch server (`pavics-compose.sh restart finch`) and clear the queue left by the previous hung process (`scripts/clear-running-wps-jobs-in-db.sh finch $POSTGRES_PAVICS_USERNAME`), to avoid the max connection problem, then only run the async process (do not run the sync process before) and it works. `$POSTGRES_PAVICS_USERNAME` is from `env.local` file.

Note it's restarting the WPS server (Finch) that unblock async request until the next sync request and async will be blocked again.  Clearing the queue is just a nice to have since old items will eventually burst the max limit of concurrent item in the queue.

## Environment

* finch version used, if any: 0.5.2
* Python version, if any: version in the Finch docker image
* Operating System: Centos 7 (production deployment)

## Additional Information

Might relate to https://github.com/bird-house/finch/issues/45

Edit:
* add missing step to clear queue also
* clarify that it's the restart that unblock async request, clearing the queue is just nice to have

**Comment by tlvu** [comment]
Upcoming "jobqueue" feature from PyWPS (https://github.com/geopython/pywps/pull/505) might help fix this.

**Comment by huard** [comment]
Running without dask resolves this problem. 
There seems to be an interaction between dask scheduler/workers and the PyWPS multiprocessing async queue. 
@cjauvin is investigating.

**Comment by cjauvin** [comment]
Related: 
* https://github.com/geopython/pywps/issues/627
* https://github.com/dask/distributed/issues/5334

**Comment by huard** [comment]
Hopefully fixed by #204 #211

**Comment by tlvu** [comment]
@eyvorchuk seems to have hit this problem again so the fix might not have worked for all cases.
